lib/utils/compose_ts.ts

```diff
@@ -775,7 +889,22 @@ async function newTarDirectory(
 				mode: fileStats.stats.mode,
 				size: fileStats.stats.size,
 			},
-			await readFile(fileStats.filePath),
+			fileStats.relPath === configPath
+				? setConfig(configPath, config, parsedBuildArgs)
+				: await readFile(fileStats.filePath),
+		);
+	}
+	if (
+		configPath === '' &&
+		(!_.isEmpty(parsedBuildArgs.global) || !_.isEmpty(parsedBuildArgs.services))
+	) {
+		const configData = setConfig('.balena/balena.yml', {}, parsedBuildArgs);
+		pack.entry(
+			{
+				name: toPosixPath('.balena/balena.yml'),
```

Comment:
`toPosixPath` is not needed here, because the given path is a constant with forward slashes.
The function is defined in `resin-multibuild` and its purpose is to replace backslashes with forward slashes as the path separator:
```
export function toPosixPath(p: string): string {
	return p.replace(nativeSepRE, '/');
}
```